llama_stack/distribution/server/server.py

```diff
@@ -95,7 +96,7 @@ def translate_exception(exc: Exception) -> Union[HTTPException, RequestValidatio
                 ]
             },
         )
-    elif isinstance(exc, ValueError):
+    elif isinstance(exc, ValueError) or isinstance(exc, AssertionError):
```

Comment:
I think assertions are still internal. They should be translated into appropriate ValueErrors at the place where you are throwing that assert -- if you think there is no internal state being divulged. The assertion message can then be approrpiately changed to indicate that this is a user-facing message (and would be nicer.)

Reply:
I was tempted to change those `assert` to `raise exception. Is there any reason why we are using `assert`? Usually assert is for fatal issue and we need to shutdown the application as I know?

Let me change those assert to raise value exception then, if appropriate.

Reply:
@chuenlok Indeed, the assert _may_ be inappropriate and perhaps we should introduce a specific `UnexpectedModelOutputError` which we then explicitly handle.

Reply:
Here's a friendly list of what exceptions are usually semantic in nature and can be caught at the highest level and potentially exposed to the user: https://gist.github.com/ashwinb/8f2ced735408774f6bd41b5188ae974c

Reply:
> Here's a friendly list of what generic Exceptions could be caught at the highest level and potentially exposed to the user: https://gist.github.com/ashwinb/8f2ced735408774f6bd41b5188ae974c

This is really useful! I am still new to the stack and might not be able to use the correct exception. Let me give it a try based on your list

Reply:
Converted some assert to exceptions and leave the internal one untouched.